### PR TITLE
[STAN] Add stats data stream metric types

### PR DESCRIPTION
**Pull request by ghost**
## What does this PR do?

Add metric types for `stats` data stream fields.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ]

## Related issues

- Relates #7859

## Screenshots
The dashboard for STAN metrics seems to be broken (this has to be addressed). I recreated the lenses just for visualization and data validation.

scrnli_9_22_2023_12-15-28 AM

### CODE REVIEW

**Comment by gpop63** [comment]
<details>
<summary>Rollover test</summary>

```json
{
  "acknowledged": true,
  "shards_acknowledged": true,
  "old_index": ".ds-metrics-stan.stats-default-2023.09.21-000001",
  "new_index": ".ds-metrics-stan.stats-default-2023.09.21-000002",
  "rolled_over": true,
  "dry_run": false,
  "conditions": {}
}
```

</details> 

<details>
<summary>Mappings</summary>

```json
"stan": {
          "properties": {
            "cluster": {
              "properties": {
                "id": {
                  "type": "keyword",
                  "time_series_dimension": true
                }
              }
            },
            "server": {
              "properties": {
                "id": {
                  "type": "keyword",
                  "time_series_dimension": true
                }
              }
            },
            "stats": {
              "properties": {
                "bytes": {
                  "type": "long",
                  "time_series_metric": "counter"
                },
                "channels": {
                  "type": "long",
                  "time_series_metric": "gauge"
                },
                "clients": {
                  "type": "long",
                  "time_series_metric": "gauge"
                },
                "messages": {
                  "type": "long",
                  "time_series_metric": "counter"
                },
                "role": {
                  "type": "keyword",
                  "ignore_above": 1024
                },
                "state": {
                  "type": "keyword",
                  "ignore_above": 1024
                },
                "subscriptions": {
                  "type": "long",
                  "time_series_metric": "gauge"
                }
              }
            }
          }
        }
```

</details>

**Comment by constanca-m** [comment]
Is the dashboard breaking related to the metric type here, or was it broken before? @gpop63

**Comment by gpop63** [comment]
@constanca-m dashboard was broken before

**Review by constanca-m — APPROVED** [review]

**Comment by gpop63** [comment]
TSDB testing kid output:

```sh
Testing data stream metrics-stan.stats-default.
Index being used for the documents is .ds-metrics-stan.stats-default-2023.10.18-000001.
Index being used for the settings and mappings is .ds-metrics-stan.stats-default-2023.10.18-000001.

The time series fields for the TSDB index are: 
        - dimension (11 fields):
                - agent.id
                - cloud.account.id
                - cloud.availability_zone
                - cloud.instance.id
                - cloud.provider
                - cloud.region
                - container.id
                - host.name
                - service.address
                - stan.cluster.id
                - stan.server.id
        - counter (2 fields):
                - stan.stats.bytes
                - stan.stats.messages
        - gauge (3 fields):
                - stan.stats.channels
                - stan.stats.clients
                - stan.stats.subscriptions
        - routing_path (11 fields):
                - agent.id
                - cloud.account.id
                - cloud.availability_zone
                - cloud.instance.id
                - cloud.provider
                - cloud.region
                - container.id
                - host.name
                - service.address
                - stan.cluster.id
                - stan.server.id

Index tsdb-index-enabled successfully created.

Copying documents from .ds-metrics-stan.stats-default-2023.10.18-000001 to tsdb-index-enabled...
All 1459 documents taken from index .ds-metrics-stan.stats-default-2023.10.18-000001 were successfully placed to index tsdb-index-enabled.
```

File: packages/stan/data_stream/stats/fields/fields.yml [hunk 1]
```diff
@@ -12,21 +12,26 @@
         If clustered, role of this node in the cluster (Leader, Follower, Candidate)
     - name: clients
       type: integer
+      metric_type: gauge
```
[the remark this answers is not in the document]
Comment (gpop63) [reply packages/stan/data_stream/stats/fields/fields.yml:15]: <b>Clients and subscriptions:</b>

<details>
<summary>Before</summary>
<p>

```json
{
   "@timestamp":"2023-10-21T17:47:37.290Z",
   "service":{
      "address":"http://stan:8222/streaming/serverz",
      "type":"stan"
   },
   "stan":{
      "server":{
         "id":"o2SjxYZjVJhoILVqjXFn90"
      },
      "cluster":{
         "id":"test-cluster"
      },
      "stats":{
         "subscriptions":4,
         "clients":4,
         "channels":3,
         "bytes":15776304,
         "messages":188031,
         "state":"STANDALONE"
      }
   }
}
```

</p>
</details> 

<details>
<summary>After closing STAN client</summary>
<p>

```json
{
   "@timestamp":"2023-10-21T17:47:47.290Z",
   "service":{
      "address":"http://stan:8222/streaming/serverz",
      "type":"stan"
   },
   "stan":{
      "server":{
         "id":"o2SjxYZjVJhoILVqjXFn90"
      },
      "cluster":{
         "id":"test-cluster"
      },
      "stats":{
         "subscriptions":3,
         "clients":3,
         "channels":3,
         "bytes":15777984,
         "messages":188051,
         "state":"STANDALONE"
      }
   }
}
```

</p>
</details> 

<b>Channels:</b>

Through testing, I found no direct method to delete a STAN channel via API. Instead, I manually deleted the channel's folder from `/data/stan/<channel-name>` and restarted STAN to apply the change.

<details>
<summary>Before</summary>
<p>

```json
{
  "cluster_id": "test-cluster",
  "server_id": "o2SjxYZjVJhoILVqjXFn90",
  "version": "0.25.5",
  "go": "go1.19.10",
  "state": "STANDALONE",
  "now": "2023-10-21T17:56:28.819900398Z",
  "start_time": "2023-10-21T17:43:18.424936206Z",
  "uptime": "13m10s",
  "clients": 2,
  "subscriptions": 2,
  "channels": 4,
  "total_msgs": 189077,
  "total_bytes": 15864166,
  "in_msgs": 1611,
  "in_bytes": 140157,
  "out_msgs": 4099,
  "out_bytes": 180354,
  "open_fds": 25,
  "max_fds": 1048576
}
```

</p>
</details> 

<details>
<summary>After</summary>
<p>

```json
{
  "cluster_id": "test-cluster",
  "server_id": "hCSE0bDHkBTTVv9iV3GBAd",
  "version": "0.25.5",
  "go": "go1.19.10",
  "state": "STANDALONE",
  "now": "2023-10-21T17:56:58.678350936Z",
  "start_time": "2023-10-21T17:56:34.919167622Z",
  "uptime": "23s",
  "clients": 2,
  "subscriptions": 2,
  "channels": 2,
  "total_msgs": 187431,
  "total_bytes": 15727676,
  "in_msgs": 0,
  "in_bytes": 0,
  "out_msgs": 2,
  "out_bytes": 96,
  "open_fds": 21,
  "max_fds": 1048576
}
```

</p>
</details>

**Review by agithomas — APPROVED** [review]
LGTM!

**Pull request merged**